lib/rubocop/cop/mixin/non_modifier_if_then.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    # Common functionality for checking if an `if` node is in nonmodifier form with implicit `then`.
+    module NonModifierIfThen
+      NON_MODIFIER_THEN = /then\s*(#.*)?$/.freeze
+
+      def non_modifier_if_then?(node, same_line: false)
+        return false unless node.if_type?
+
+        if same_line
```

Comment:
I'm a bit puzzled about this check as in the two branches you seems to be checking pretty different things. What is the `same_line` check supposed to do? (without tests it's a bit hard to be sure what you intended to achieve here)